doc/decisions/DR016-One-to-many-result-paging.md

```diff
@@ -0,0 +1,669 @@
+            """, my_relationship_type, str(input))
+
+        pageToken = PageToken(cursor)
+
+    if limit <= 0:
+        result_set = pageToken.cursor.fetchall()
+    else:
+        result_set = pageToken.cursor.fetchmany(limit)
+
+    related_refs = [EntityReference(result[0]) for result in result_set]
+    return pageToken, related_refs
+```
+
+#### Page token alternative - command pattern
+
+One of the major Cons of page tokens as sketched so far is that the
+original query arguments must be redundantly re-specified when fetching
+subsequent pages. To overcome this we can split the construction of page
+tokens from the operations performed on them.
+
+This could be done as a series of methods, however that then looks
+extremely similar to a cursor object design.
+
+Instead, the sketch below takes a command pattern approach. We add a
+single new method, with the operation to perform given as an additional
+argument.
+
+```python
+def relatedEntities(
+        input: EntityReference, relationship) -> AbstractPageToken:
+
+    my_relationship_type: str = get_my_relationship_type(relationship)
+
+    cursor = get_my_connection().cursor()
+    cursor.execute(
+        f"""
+        SELECT ToRef FROM Relationship
+        WHERE RelationshipType = ?
+        AND FromRef = ?
+        """, my_relationship_type, str(input))
+
+
+def relatedEntitiesPageCmd(
+    pageToken, command="next", limit=-1
+) ->  Tuple[AbstractPageToken, List[EntityReference]]:
+
+    if command == "next":
+        if limit <= 0:
+            result_set = pageToken.cursor.fetchall()
+        else:
+            result_set = pageToken.cursor.fetchmany(limit)
+
+        related_refs = [EntityReference(result[0]) for result in result_set]
+
+        return pageToken, related_refs
+    else:
+        raise NotImplementedError()
+```
+
+#### Cursor object signature
+
+The following sketch wraps a DB-API cursor in an `AbstractCursor`,
+showing a straightforward mapping of functionality, and demonstrating
+the extensibility to support other common cursor API methods (in this
+case, skipping over records without the cost of fetching all the
+results).
+
+```python
+class Cursor(AbstractCursor):
+    def __init__(self, cursor):
+        self.__cursor = cursor
+
+    def getPage(self, pageSize) -> List[EntityReference]:
+        result_set = self.__cursor.fetchmany(pageSize)
+        return [
+            EntityReference(result) for result[0] in result_set]
+
+    def skip(self, pageSize):
+        self.__cursor.scroll(pageSize)
+
+
+def relatedEntities(
+        input: EntityReference, relationship) -> AbstractCursor:
+    my_relationship_type: str = get_my_relationship_type(relationship)
+
+    cursor = get_my_connection().cursor()
+    cursor.execute(
+        f"""
+        SELECT ToRef FROM Relationship
+        WHERE RelationshipType = ?
+        AND FromRef = ?
+        """, my_relationship_type, str(input))
+
+    return Cursor(cursor)
+```
+
+### Using a REST API
```

Comment:
After offline conversation, added a sketch using a theoretical REST API in 22c9f811